src/vivarium/framework/values.py

```diff
@@ -170,6 +172,29 @@ def union_post_processor(values: List[NumberLike], _) -> NumberLike:
     return joint_value
 
 
+def step_size_post_processor(values: List[NumberLike], _) -> pd.Series:
+    """Computes the largest feasible step size for each simulant. This is the smallest component-modified
+    step size, or the global step size, whichever is larger. If no components modify the step size, we default
+    to the global step size.
+
+    Parameters
+    ----------
+    values
+        A list of step sizes
+
+    Returns
+    -------
+    pandas.Series
+        The largest feasible step size for each simulant
+
+
+    """
+    if len(values) == 1:
+        return values[0]
+    min_modified = pd.DataFrame(values[1:]).min(axis=0).astype("timedelta64[ns]")
```

Comment:
I think this doesn't quite work, since `values` is a list of `pd.Series` -- we'd need something like an iteration over the items in the series somewhere to get a row-wise minimum. I think it's fairly natural to think of a `List[pd.Series]` as basically a `DataFrame`, which we are squeezing down using `.min()` but YMMV. I will add a comment that makes this a bit more explicit.